.github/workflows/flowzone.yml

```diff
@@ -387,6 +389,111 @@ jobs:
+          else
+            echo "enabled=false" >> $GITHUB_OUTPUT
+          fi
+
+      # Check which Python 3.7+ versions meet the Poetry project requirements
+      - name: Install Poetry
+        if: steps.python_poetry.outputs.enabled == 'true'
+        run: |
+          pipx install poetry
+
+      - name: Set up Python 3.7
+        if: steps.python_poetry.outputs.enabled == 'true'
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.7"
+
+      - name: Check compatibility
+        if: steps.python_poetry.outputs.enabled == 'true'
+        run: |
+          error_check=`(poetry env use 3.7 2>&1 || true)`
+          if ! grep -q "Please choose a compatible version" <<< $error_check
+          then
+            echo "PYTHON_VERSIONS=$(echo "${PYTHON_VERSIONS}" | jq -c '. + ["3.7"]')" >> $GITHUB_ENV
+          else
+            echo "Python 3.7 does not meet project requirements."
+          fi
+
+      - name: Set up Python 3.8
+        if: steps.python_poetry.outputs.enabled == 'true'
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.8"
+
+      - name: Check compatibility
+        if: steps.python_poetry.outputs.enabled == 'true'
+        run: |
+          error_check=`(poetry env use 3.8 2>&1 || true)`
+          if ! grep -q "Please choose a compatible version" <<< $error_check
+          then
+            echo "PYTHON_VERSIONS=$(echo "${PYTHON_VERSIONS}" | jq -c '. + ["3.8"]')" >> $GITHUB_ENV
+          else
+            echo "Python 3.8 does not meet project requirements."
+          fi
+
+      - name: Set up Python 3.9
+        if: steps.python_poetry.outputs.enabled == 'true'
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.9"
+
+      - name: Check compatibility
+        if: steps.python_poetry.outputs.enabled == 'true'
+        run: |
+          error_check=`(poetry env use 3.9 2>&1 || true)`
+          if ! grep -q "Please choose a compatible version" <<< $error_check
+          then
+            echo "PYTHON_VERSIONS=$(echo "${PYTHON_VERSIONS}" | jq -c '. + ["3.9"]')" >> $GITHUB_ENV
+          else
+            echo "Python 3.9 does not meet project requirements."
+          fi
+
+      - name: Set up Python 3.10
+        if: steps.python_poetry.outputs.enabled == 'true'
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Check compatibility
+        if: steps.python_poetry.outputs.enabled == 'true'
+        run: |
+          error_check=`(poetry env use 3.10 2>&1 || true)`
+          if ! grep -q "Please choose a compatible version" <<< $error_check
+          then
+            echo "PYTHON_VERSIONS=$(echo "${PYTHON_VERSIONS}" | jq -c '. + ["3.10"]')" >> $GITHUB_ENV
+          else
+            echo "Python 3.10 does not meet project requirements."
+          fi
+
+      # default to the latest version (^3.7) on the runner 
+      # if none were matched by the checks above
```

Comment:
Not really, it does not check what [versions](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md#python) the runner provides. I guess this can be improved along with the comment above and applied to Node and npm tests as well.